### DISCUSSION TITLE
Use Cy.log within app source code?

### DISCUSSION THREAD

**Discussion by dclowd9901** [cypress-io/cypress#15945] [Questions and Help]
Hi,

I've run into an issue writing a test where, when I open the console and watch output from `console.log`s in the source code, it fails, because it slows the test down too much and causes it to time out. 

I was thinking it would be nice if I could just `cy.log` from the source code, and see the output in the cypress stdout logger, but I've tried doing this by putting `cy.log` on the `window` object, but it never seems to be available to the source code. I know this is pretty janky, but if anyone has a way I might be able to do this, or something like it, I'd love to hear it.

**Comment by EtienneBruines** [comment]
I´ve been successfully using this:

```js
      window.Cypress.log({ name: 'hello', message: 'world' })
```

This way it at least pops up in the runner (not stdout), which means it is captured during `cy.screenshot` and videos. Alternatively you could write a task that logs it to stdout instead, available under `window.Cypress`.

**Comment by dclowd9901** [comment]
Thank you for the response. I’ll try this out.

**Reply by lmiller1990** [reply]
Did this work for you?

I don't fully understand what you mean by "times out". `console.log` should not introduce any overhead, right?

If you can provide a bit more info, I can help!
